totem/Tester.lua

```diff
@@ -734,13 +734,10 @@ function Tester:run(tests)
     else
         status = self:_run(self:_getTests(tests))
     end
-    -- Detect whether we're running directly from a top-level script. This
-    -- will not work for e.g. th which is an interpreter defined in lua
-    if debug.getinfo(3) then
-        return status
-    else
-        os.exit(status, true)
-    end
+    -- Throws an error on test failure/error, so that test script returns
+    -- with nonzero return value.
+    assert(status == 0, 'An error was found while running tests!')
```

Comment:
this *almost* works. But for the script totem-run to still work, need to change one more thing. In line 923 below, instead of self:add(dofile(f), f)  (which will now fail, because dofile(f) will throw an error) you need to do the following:
```
local success, res = pcall(dofile, f)
self:add((success and res) or -1, f)
```
to make sure that the thrown error is caught and translated into a -1 return value, which will then be correctly used to report the overall test results.